File: src/main/java/com/example/BookStore/model/Publisher.java
package com.example.BookStore.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "publisher")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Publisher {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    @OneToMany(mappedBy = "publisher")
    private Set<Author> authors ;
    @OneToMany(mappedBy = "publisher")
    private Set<Book> books;
    private String address;
    private String contactDetails;
}

File: src/main/java/com/example/BookStore/model/Publisher.kt
package com.example.BookStore.model

import jakarta.persistence.*


@Entity
@Table(name = "publisher")
data class Publisher(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id:Long,
    var name:String,
    @OneToMany(mappedBy = "publisher")
    var authors: MutableSet<Author>? = mutableSetOf<Author>(),
    @OneToMany(mappedBy = "publisher")
    var books: MutableSet<Book>? = mutableSetOf<Book>(),
    var address:String?,
    var contactDetails: String?

)
